sdk/python/feast/infra/gcp.py

```diff
@@ -27,6 +26,14 @@
 from feast.registry import Registry
 from feast.repo_config import DatastoreOnlineStoreConfig, RepoConfig
 
+try:
+    from google.auth.exceptions import DefaultCredentialsError
+    from google.cloud import bigquery, datastore
```

Comment:
iirc we didn't have google cloud dependencies imported at the top because Feast will try to import them even when the user is not using gcp (e.g. they're just trying things locally) due to this file being imported somewhere else. If you uninstall the google cloud pip packages in your venv and try the quickstart you can confirm whether this happens

Reply:
Yea, but I think its cleaner to conditionally import our providers and stores than their subdependencies. Let me test to validate it.

Reply:
I think only this [change](https://github.com/feast-dev/feast/pull/1561/commits/27040ec0bfa28485a5133f1d6180bb997db045a6) was necessary. Our tests are failing for other reasons